internal/tests/api/wallet_test.go

```diff
@@ -83,6 +86,61 @@ func TestWalletRecover_WrongDerivationPath(t *testing.T) {
 	t.Logf("Wallet address doesn't match as expected (expected %s, got %s)", expectedWalletAddress.Hex(), response.Data.AccountAddress.Hex())
 }
 
+func TestWalletStatus_NotLoaded(t *testing.T) {
+	apiClient := testMgr.GetApiClient()
+	response, err := apiClient.Wallet.Status()
+	require.NoError(t, err)
+	t.Log("Status called")
+
+	require.Equal(t, emptyWalletAddress, response.Data.WalletStatus.Address.NodeAddress)
+	require.False(t, response.Data.WalletStatus.Address.HasAddress)
+
+	require.Equal(t, wallet.WalletType(""), response.Data.WalletStatus.Wallet.Type)
+	require.False(t, response.Data.WalletStatus.Wallet.IsLoaded)
+	require.False(t, response.Data.WalletStatus.Wallet.IsOnDisk)
+	require.Equal(t, emptyWalletAddress, response.Data.WalletStatus.Wallet.WalletAddress)
+
+	t.Log("Received correct wallet status")
+}
+
+func TestWalletStatus_Loaded(t *testing.T) {
+	// Take a snapshot, revert at the end
+	snapshotName, err := testMgr.CreateCustomSnapshot(osha.Service_EthClients | osha.Service_Filesystem)
+	if err != nil {
+		fail("Error creating custom snapshot: %v", err)
+	}
+	defer wallet_cleanup(snapshotName)
+
+	// Commit a block just so the latest block is fresh - otherwise the sync progress check will
+	// error out because the block is too old and it thinks the client just can't find any peers
+	err = testMgr.CommitBlock()
+	if err != nil {
+		t.Fatalf("Error committing block: %v", err)
+	}
+
+	// Regen the wallet
+	derivationPath := string(wallet.DerivationPath_Default)
+	index := uint64(0)
+	apiClient := testMgr.GetApiClient()
+	_, err = apiClient.Wallet.Recover(&derivationPath, keys.DefaultMnemonic, &index, goodPassword, true)
+	require.NoError(t, err)
+	t.Log("Recover called")
+
+	response, err := apiClient.Wallet.Status()
+	require.NoError(t, err)
+	t.Log("Status called")
+
+	require.Equal(t, expectedWalletAddress, response.Data.WalletStatus.Address.NodeAddress)
+	require.True(t, response.Data.WalletStatus.Address.HasAddress)
+
+	require.Equal(t, wallet.WalletType("local"), response.Data.WalletStatus.Wallet.Type)
```

Comment:
For the expected val, you should use the actual type (`wallet.WalletType_Local`) instead of casting a string.

Reply:
fixed